### ISSUE TITLE
Eject option

### ISSUE THREAD

**Issue by azdanov** [jorgegonzalez/zoe#13]
Something similar to [create-react-app](https://github.com/facebook/create-react-app)'s `eject` option. Which will setup necessary configuration files for eslint and prettier with default settings and dev-dependencies.

Reason for this is native linting extensions for `visual studio code` or `webstorm` won't go into `node_modules`_(could be mistaken)_.
And also personal preference, to catch errors as I type.

**Comment by jorgegonzalez** [comment]
Do you think this is still a good feature to have, considering users can now use the shareable config? Not sure what problem this would solve now.

**Comment by azdanov** [comment]
When using _Intellij_ there needs to be an `.estlintrc`, and `.prettierrc` for the plugins to start working. I can manually configure them, for each project. Eject would solve all that file creation / configuration.

Just typing `zoe eject` would create necessary: `.zoerc`, `.eslintrc`, `.prettierrc`, with the default config, and all the plugins / configs already set in `package.json`.

Maybe I'm overthinking this, since zoe is a linter / formatter.

**Comment by jorgegonzalez** [comment]
> Maybe I'm overthinking this, since zoe is a linter / formatter.

No, I can see how it would be useful to generate those.

Just curious, does Intellij need those files even if they're empty?

**Comment by azdanov** [comment]
Yeah, but not empty. They still a minimal config to use. That's certain for eslint, prettier could be omitted since it already includes defaults.

**Comment by jorgegonzalez** [comment]
`zoe --eject` yes?

**Comment by azdanov** [comment]
Should be `zoe eject` since it's a command not a modifier. Same as `zoe init`.

**Comment by jorgegonzalez** [comment]
Currently `init` is implemented as `zoe --init`.

**Comment by jorgegonzalez** [comment]
Should they both be commands? What if a user runs `zoe` on a directory named `init` or `eject`? Perhaps that's a rare edgecase?

**Comment by azdanov** [comment]
You're right. Prefixing with `--` is safer.

**Comment by jorgegonzalez** [comment]
I'm wondering if we can use this option to modify existing `.eslintrc` files. Maybe that's a step in the right direction for better editor integration?

**Comment by jorgegonzalez** [comment]
> I used to use xo but the main issue we found was that editor integration was a bit tricky. Lots of plugins had to be created specifically for xo, for example. If you sort out that issue, then it'll scale really well!

I originally intended to write a Zoe plugin for VSCode. But, as described in this feedback, we could run into the same problem XO had, in that there would need to be a specific plugin per editor.

A wiser move would be to depend on each existing editor's native ESLint plugin.

**Comment by jorgegonzalez** [comment]
Perhaps when a user runs `zoe --init`, we can do something like this:

```
$ zoe --init

Looks like you're running zoe for the first time.
We've detected an existing ".eslintrc" file.
Can we modify it for you?
```

**Comment by jorgegonzalez** [comment]
For plugins, `.eslintrc` files would be more important, so what would be the point of having `.zoerc` files? Not sure how to reconcile these two.

**Comment by jorgegonzalez** [comment]
https://github.com/Microsoft/vscode-eslint/issues/498

**Comment by azdanov** [comment]
> I agree, we're on track and probably better off combining both.

I see it this way: 
* One module is responsible for configuration and user inquiry, `zoe-config`, about `init` of a project and maybe `eject`. 
* Another is a wrapper around eslint, prettier and whatever tools (stylelint?) might be added, `zoe-lint`, which gives access with simple commands, `zoe` and `zoe --fix`.

This way they don't collide, since they have separate concerns, and are easier to change. Both are required inside the _main_ `zoe` app, instead of trying to squeeze everything together.

This will need some moving of the code around, yet sure is worth it in the long run.
